### ISSUE TITLE
Support for more architecture and functions

### ISSUE THREAD

**Issue by annopackage** [microsoft/nni#2335]
**Short summary about the issue/question:**

**Brief what process you are following:**
1. Does the repo support for prunning which contains short cut and concatenation operation now? 
2. How to get the architecture after prunning from the mask? Is there any function directly be used for this? 

**How to reproduce it:**

nni Environment:

nni version:1.4
nni mode(local|pai|remote):
OS:window/liunx
python version:3.6
is conda or virtualenv used?: conda
is running in docker?:no
need to update document(yes/no):

Looking forward to your reply. Thanks in advance.

**Anything else we need to know:**

**Comment by annopackage** [comment]
From the given examples, we could only get the best prunning mask or sparse model weights. It seems that modelspeedup could help get the pruned model architecture?

**Comment by chicm-ms** [comment]
Hi @annopackage ,
Thanks for being interested in nni.

1. NNI supports model pruning with 2 steps:

Step1:  Pruning a model with a pruner, such as SlimPruner, L1FilterPruner, in this step, models with short cut and concatenation are supported.  Each algo supports pruning different type of layers. 

Step2: Speedup the pruned model with the masks obtained in step 1, in this step, by now, nni does not support unstructured masks yet, it will be supported in future release.  For structured algos such as L1FilterPruner, nni does not support short cut and concatenation yet. These will be added later.

2. Yes, you can get pruned architecture using modelspeedup.

**Comment by annopackage** [comment]
Hi, thanks for your reply.  Currently, it only supports group=1 in conv2d. Do you have plans to support it later?

**Comment by annopackage** [comment]
I also have another question : If we fine tune the pruned mode based on the mask, I wonder the fine-tuned weight we get is sparse or not.

**Comment by chicm-ms** [comment]
@annopackage ,
Yes, conv2d groups>1 for modelspeedup will be supported later.
The finetuned weight is sparse weight.

**Comment by annopackage** [comment]
OK, looking forward to your update.

**Comment by QuanluZhang** [comment]
@annopackage your issue may be resolved by pr #2579
